src/service/impl/stops/departure-group.ts

```diff
@@ -122,7 +130,40 @@ export async function getDeparturesGrouped(
   options: DepartureGroupsQuery,
   favorites?: FavoriteDeparture[]
 ): Promise<Result<DepartureGroupMetadata, APIError>> {
-  const ids = Array.isArray(id) ? id : [id];
+  let ids = Array.isArray(id) ? id : [id];
+
+  console.log('getDeparturesGrouped ');
+
+  // Some stops can be multi modal, meaningstops can have children
+  // stops which again has quays. This checks if given stop is multimodal
+  // and if it is, fetch all other stops that are
+  // children of that stop place _or_ the parent stop place..
+  const expandedMultiModalStops = await Promise.all(
+    ids.map(async function (stopId) {
+      const potentialChildrenResult = await stopPlacesClient.query<
+        QuaysInMultimodalQuery,
+        QuaysInMultimodalQueryVariables
+      >({
+        query: QuaysInMultimodalDocument,
+        variables: { stopId }
+      });
+
+      // If no data or error, just treat it as normal stop place.
+      // Data will stil show but not perfectly.
+      const data = potentialChildrenResult.data?.stopPlace;
+      if (potentialChildrenResult.errors || !data) {
+        return [stopId];
+      }
+
+      // Find all children or just default to itself.
+      const ret = (data[0]?.children
+        ?.map(c => c.id)
+        .filter(Boolean) as string[]) ?? [stopId];
+      return ret;
```

Comment:
Can `data[0]?.children?.map(c => c.id).filter(Boolean)` end up as empty array here? And if so, is it intended that the empty array is returned as `[] ?? [stopId]` gives `[]`. 

If empty array never should be returned, then maybe something like this feels safer:
```suggestion
      const childrenStopIds = data[0]?.children?.map(c => c.id).filter(Boolean);
      return childrenStopIds?.length ? (childrenStopIds as string[]) : [stopId];
```